connectors/crowdstrike/connector/server_access.py

```diff
@@ -60,7 +60,7 @@ def get_hosts(self, last_model_state_id=None):
                     for err in error_detail:
                         raise DatasourceFailure(err["message"], err["code"])
                 if len(hosts) < host_lookup['body']['meta']['pagination']['total']:
-                    offset = len(hosts)
+                    offset = host_lookup['body']['meta']['pagination']['offset']
```

Comment:
use python dictionary get() method to avoid the key errors. https://docs.python.org/3/library/stdtypes.html#dict.get
since we are unsure about the offset value we need to use this method to get the correct value and handle the exception as well.